### PR TITLE
chore(parquet): track and use parquet read size in queries

### PR DESCRIPTION
**Pull request by amunra**
The O3 logic for parquet partitions can update a file in place by appending new replacement rowgroups and a new metadata section at the end of the file.
This means that readers (such as queries) can still access the old state by accessing the parquet partitions at an older file size.
When reading parquet, we thus need to track a specific valid file size to avoid:
* Accidentally picking up a new state.
* Accidentally reading an incomplete file with partially appended data that is in the process of being appended to.

### CODE REVIEW

**Comment by questdb-butler** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please update the PR title to match this format:
`type(subType): description`

Where `type` is one of:
`feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, `test`, `ci`, `revert`

And: `subType` is one of:
`build`, `sql`, `log`, `mig`, `core`, `ilp`, `pgwire`, `http`, `conf`, `ui`, `wal`, `parquet`

For Example:

```
perf(sql): improve pattern matching performance for SELECT sub-queries
```
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8eb6399eae9c85faf1723a7c390c0749f309af9a
</p>

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 80 / 90 (88.89%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/FullBwdPartitionFrameCursor.java|0|4|00.00%|
|:large_blue_circle:|io/questdb/cairo/AbstractIntervalPartitionFrameCursor.java|1|5|20.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/BwdTableReaderPageFrameCursor.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/griffin/engine/table/parquet/PartitionDecoder.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/ReadParquetRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|parquet_read/decode.rs|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/FwdTableReaderPageFrameCursor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/ReadParquetFunctionFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/TxReader.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|
|:large_blue_circle:|parquet_read/meta.rs|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/SelectedRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageFrameMemoryPool.java|3|3|100.00%|
|:large_blue_circle:|parquet_read/jni.rs|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageFrameAddressCache.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/AbstractFullPartitionFrameCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/FullFwdPartitionFrameCursor.java|3|3|100.00%|
|:large_blue_circle:|parquet_read/mod.rs|2|2|100.00%|

File: core/src/main/java/io/questdb/cairo/TableWriter.java [hunk 1]
```diff
@@ -1421,7 +1421,8 @@ public boolean convertPartitionParquetToNative(long partitionTimestamp) {
         // path is now pointing to the parquet file
         // other is pointing to the new partition folder
         LOG.info().$("converting parquet partition to native [path=").$substr(pathRootSize, path).I$();
-        long parquetFd = TableUtils.openRO(ff, path.$(), LOG);
+        final long parquetFd = TableUtils.openRO(ff, path.$(), LOG);
+        final long readSize = ff.length(parquetFd);
```
Comment (amunra) [inline core/src/main/java/io/questdb/cairo/TableWriter.java:1425]: Probably not correct.

Comment (amunra) [reply core/src/main/java/io/questdb/cairo/TableWriter.java:1425]: The table writer here should have exclusive access to the partition.
What if we introduce a partition compactor job later?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 82 / 88 (93.18%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/FullBwdPartitionFrameCursor.java|0|4|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/BwdTableReaderPageFrameCursor.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/griffin/engine/table/parquet/PartitionDecoder.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/ReadParquetRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|parquet_read/decode.rs|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/FwdTableReaderPageFrameCursor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/ReadParquetFunctionFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/IntervalFwdPartitionFrameCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/TxReader.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionJob.java|1|1|100.00%|
|:large_blue_circle:|parquet_read/meta.rs|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/SelectedRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/ParquetTimestampFinder.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageFrameMemoryPool.java|3|3|100.00%|
|:large_blue_circle:|parquet_read/jni.rs|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageFrameAddressCache.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/AbstractFullPartitionFrameCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/FullFwdPartitionFrameCursor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/AbstractIntervalPartitionFrameCursor.java|2|2|100.00%|
|:large_blue_circle:|parquet_read/mod.rs|2|2|100.00%|

**Comment by puzpuzpuz** [comment]
Merged into #5051

**Pull request closed without merging**